### ISSUE TITLE
[7.0] sending mail fails

### ISSUE THREAD

**Issue by legalsylvain** [odoo/odoo#8409]
Hi,
I have a bug, sending emails.
I suspect this commit #3848. (Merged in July 2015)

# Step to reproduce
- Install a fresh Odoo V7 version with sale ; 
- set correctly outgoing mail settings ;
- create a new quotation ; 
- click on "send by email", and click on "send" ; 

# result 
- the mail is not sent ;
- a silent error is logged ;
```
    2015-09-07 ERROR test__mail openerp.addons.mail.mail_mail: failed sending mail.mail 25
    Traceback (most recent call last):
    File "/Odoo/addons/mail/mail_mail.py", line 325, in send
        context=context
    File "/Odoo/openerp/addons/base/ir/ir_mail_server.py", line 426, in send_email
        "It should contain one valid plain ASCII email") % smtp_from
    AssertionError: Malformed 'Return-Path' or 'From' address: '"object.user_id.email or \'\'" <${>' - It should contain one valid plain ASCII email
```
# analyse

It seems that the problem is that email_from is now "${(object.user_id.email or '')|safe}", and this value is not evaluated;

Sorry I don't know how to write a unittest that fails, because of mail configuration, but I'll provide any extra informations you could need.

Maybe my analysis is not correct. 
But for my side, I just set "If False" on this line, to fix the problem.
https://github.com/OCA/OCB/commit/0f82346167698fb7349ff16618d1cecf8984cca9#diff-a64b6641231781059a38d9c282b3b97bR159

CC : @eLBati, @antonylesuisse, @tde-banana-odoo, @mart-e

**Comment by mart-e** [comment]
Hello Sylvain,

You are right, the commit 0f82346167698fb7349ff16618d1cecf8984cca9 is indeed introducing a regression for email_from that needs to be evaluated.
I have made a fix at #8421 that should fix that.

**Comment by mart-e** [comment]
fixed at 909e6e3, thanks for the notive

**Comment by legalsylvain** [comment]
Thanks for your reactivity and for the patch provided.

regards.
